install/kubernetes/cilium/values.yaml

```
@@ -20,9 +20,13 @@ rbac:
 imagePullSecrets:
 # - name: "image-pull-secret"
 
-# kubeConfigPath: ~/.kube/config
-# k8sServiceHost:
-# k8sServicePort:
+# -- (string) Kubernetes config path
+# @default -- ~/.kube/config
+kubeConfigPath: ~
```

Since we're using the `if value` pattern here, I think using empty string (`""`) instead of `~` for those string values is more straight forward.

Let's only use nil for values where we do an explicit nil check.

---

I thought `if value` is to do `nil` check, but I guess you mean ` != nil` or `== nil` is to do `nil` check in `pkg/option/config.go`? can you give an example ? :)

---

Yes, ``{{ if value }}`` does a nil check. But what I'm proposing is that you set all string types to empty string instead of nil, because `""` will behave exactly like "~" for `{{ if value }}`. 

For example:

```suggestion
kubeConfigPath: ""
```

The `nil` values are mostly useful where `{{ if value }}` cannot be used (e.g. everything in your third commit).

---

I think I got it, thanks!